gcloud-java-contrib/gcloud-java-nio/src/test/java/com/google/gcloud/storage/contrib/nio/ITGcsNio.java

```diff
@@ -0,0 +1,360 @@
+  private static final String BUCKET = RemoteGcsHelper.generateBucketName();
+  private static final String SML_FILE = "tmp-test-small-file.txt";
+  private static final int SML_SIZE = 100;
+  // it's big, relatively speaking.
+  private static final String BIG_FILE = "tmp-test-big-file.txt";
+  // arbitrary size that's not too round.
+  private static final int BIG_SIZE = 2024 * 1024 - 50;
+  private static final String PREFIX = "tmp-test-file";
+  private static Storage storage;
+  private static StorageOptions storageOptions;
+
+  private final Random rnd = new Random();
+
+  @BeforeClass
+  public static void beforeClass() throws IOException {
+    // loads the credentials from local disk as par README
+    RemoteGcsHelper gcsHelper = RemoteGcsHelper.create();
+    storageOptions = gcsHelper.options();
+    storage = storageOptions.service();
+    // create and populate test bucket
+    storage.create(BucketInfo.of(BUCKET));
+    fillFile(storage, SML_FILE, SML_SIZE);
+    fillFile(storage, BIG_FILE, BIG_SIZE);
+  }
+
+  @AfterClass
+  public static void afterClass() throws ExecutionException, InterruptedException {
+    if (storage != null && !RemoteGcsHelper.forceDelete(storage, BUCKET, 5, TimeUnit.SECONDS)) {
+      if (log.isLoggable(Level.WARNING)) {
+        log.log(Level.WARNING, "Deletion of bucket {0} timed out, bucket is not empty", BUCKET);
+      }
+    }
+  }
+
+  public static void fillFile(Storage storage, String fname, int size) throws IOException {
+    try (WriteChannel writer = storage.writer(BlobInfo.builder(BUCKET, fname).build())) {
+      byte[] bytes = new byte[size];
+      Random rnd = new Random(size);
+      rnd.nextBytes(bytes);
+      writer.write(ByteBuffer.wrap(bytes));
+    }
+  }
+
+  @Test
+  public void testFileExists() throws IOException {
+    CloudStorageFileSystem testBucket = getTestBucket();
+    Path path = testBucket.getPath(SML_FILE);
+    assertThat(Files.exists(path)).isTrue();
+  }
+
+  @Test
+  public void testFileSize() throws IOException {
+    CloudStorageFileSystem testBucket = getTestBucket();
+    Path path = testBucket.getPath(SML_FILE);
+    assertThat(Files.size(path)).isEqualTo(SML_SIZE);
+  }
+
+  @Test
+  public void testReadByteChannel() throws IOException {
+    CloudStorageFileSystem testBucket = getTestBucket();
+    Path path = testBucket.getPath(SML_FILE);
+    long size = Files.size(path);
+    SeekableByteChannel chan = Files.newByteChannel(path, StandardOpenOption.READ);
+    assertThat(chan.size()).isEqualTo(size);
+    ByteBuffer buf = ByteBuffer.allocate(1024 * 1024);
+    int timeout = 1024 * 1024;
+    int read = 0;
+    while (chan.isOpen()) {
+      if (timeout-- <= 0) {
+        break;
+      }
+      int rc = chan.read(buf);
+      if (rc < 0) {
+        // EOF
+        break;
+      }
+      buf.clear();
+      assertThat(rc).isGreaterThan(0);
+      read += rc;
+      assertThat(chan.position() == read);
+    }
+    assertThat(read).isEqualTo(size);
```

Comment:
The reason is historical since this test was written before the file content was fixed.

Checking the contents is a good idea, so I changed the code to do just that. Thanks!